### PR TITLE
pkg: relocate functions from cli repository

### PR DESCRIPTION
**Pull request by karencfv**
## Description
This PR relocates a file from the [ecctl](https://github.com/elastic/ecctl/blob/master/pkg/util/response.go) repository which should be part of the SDK as other repositories use it.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: pkg/util/response/response.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// SetRawJSON prepares the API transport to send raw JSON
+//
+// While setting the global JSONMime producer to runtime.TextProducer() looks
+// scary, runtime.TextProducer() falls back on swag.WriteJSON(data) if the
+// object we're marshalling is a struct or slice. So it actually works in most
+// cases.
+//
+// This is only necessary because the Cloud Swagger Spec lists some JSON blob
+// data as strings. When those are changed to JSON Objects, this can go away.
+func SetRawJSON(c *client.Rest) {
```
Comment (ppapapetrou76) [inline pkg/util/response/response.go]: Is is too much to ask for a couple of UTs here? 😊

Comment (karencfv) [reply pkg/util/response/response.go]: Added some small tests :D 

In order to be able to test every line I would have to create some  functions to mock the rest client in the `mock` package. This work, imo, falls out of the scope of this PR which is part of the cloud-cli layout structure change issue. To be honest, only these two functions would need said mocks and these are already tested [here](https://github.com/elastic/cloud-cli/blob/master/app/deployment/elasticsearch/metadata/raw_test.go#L111).

Unless I'm missing some super simple way to further test this?

File: pkg/util/response/response_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		args args
+	}{
+		{
+			name: "succeeds",
+			args: args{params: &client.Rest{}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) { SetRawJSON(tt.args.params) })
+	}
```
Comment (marclop) [inline pkg/util/response/response_test.go]: I think we should at least assert that `t.Producers[runtime.JSONMime]` is equal to `runtime.TextProducer()`. I think we should use `reflect.DeepEqual` to assert the result though.

File: pkg/util/response/response_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		name string
+		args args
+	}{
+		{
+			name: "succeeds",
+			args: args{params: &client.Rest{}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) { UnsetRawJSON(tt.args.params) })
```
Comment (marclop) [inline pkg/util/response/response_test.go]: Same here, I think we should assert that `t.Producers[runtime.JSONMime]` has been set to `runtime.JSONProducer()`. I think we should use `reflect.DeepEqual` to assert the result though.

File: pkg/util/response/response.go (hunk 1 above)
Comment (marclop) [reply pkg/util/response/response.go]: If you pass a `&client.Rest{Transport: &httptransport.Runtime{}}`, you can then assert that the values of it have been modified, that's the only test needed here

Comment (karencfv) [reply pkg/util/response/response.go]: so erm, I get a panic when I try to pass `&client.Rest{Transport: &httptransport.Runtime{}}`

```
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 9 [running]:
testing.tRunner.func1(0xc000490300)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x17f9080, 0x19fc9d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/elastic/cloud-sdk-go/pkg/util/response.SetRawJSON(0xc0000ee6c0)
	/Users/karen/go/src/github.com/elastic/cloud-sdk-go/pkg/util/response/response.go:39 +0xa0
...
```

Comment (marclop) [reply pkg/util/response/response.go]: the `Producers` field (map) needs to be populated, it's complaining that an assignment is being made to a nil map, which it is (`panic: assignment to entry in nil map`).

**Comment by karencfv** [comment]
Had a chat with @ppapapetrou76 and we're going to leave this file where it is in the `cloud-cli` repo. The only package that's using these functions is the `elasticsearch/metadata` package in cloud-cli, and the functions are being tested through those tests. 😄

**Pull request closed without merging**